### ISSUE TITLE
CUPS driver usage kind of broken on v1.3

### ISSUE THREAD

**Issue by ragingdave** [rawilk/laravel-printing#16]
### Description
When attempting to use CUPS in v1.3, it seems that it's usage is broken in a few ways:
1. Just installing the package results in errors (Issue #11, partially fixed by #14 )
2. When applying the fix from #14, then there is a deprecation notice when just even attempting to list printers.

```
<warning>PHP Deprecated:  Passing a Psr\Http\Message\ResponseFactoryInterface and a Psr\Http\Message\StreamFactoryInterface to SocketClient is deprecated, and will be removed in 3.0, you should only pass config options. in /var/www/html/project/vendor/php-http/socket-client/src/Client.php on line 59</warning>

```

3. Beyond the deprecation notice, listing printers also fails due to the following exception:

```
TypeError: Argument 2 passed to Rawilk\Printing\Drivers\Cups\Cups::remoteServer() must be of the type string, null given, called in /var/www/html/project/vendor/rawilk/laravel-printing/src/Factory.php on line 46
```
This error cryptically is trying to say that a username/password are required, but the defaults for the username and password in the published config don't have a default and become null which generates the error.

Once I set the username/password to be blank strings, I then got the following error:
```
Rawilk\Printing\Exceptions\InvalidDriverConfig with message 'Remote CUPS server requires a username and password.'
```
~~This seems to be due to a arbitrary requirement of needing a username and password for a remote cups server. 
While not great security wise to have no username/password when I remove the conditional that validates for requiring username/password, the CUPS calls work just fine (albeit with the deprecation notices).~~

### Steps to reproduce

Install the package, configure only a valid CUPS_SERVER_IP, run `Printing::printers()`
Additionally, setting a username/password results in similar errors minus the error on requiring a username/password

**Context**
- Laravel Printing version: 1.3
- Laravel version: 8.35.1

**Comment by MuhammadAbbasAkhtar** [comment]
> ### Description
> When attempting to use CUPS in v1.3, it seems that it's usage is broken in a few ways:
> 
> 1. Just installing the package results in errors (Issue #11, partially fixed by #14 )
> 2. When applying the fix from #14, then there is a deprecation notice when just even attempting to list printers.
> 
> ```
> <warning>PHP Deprecated:  Passing a Psr\Http\Message\ResponseFactoryInterface and a Psr\Http\Message\StreamFactoryInterface to SocketClient is deprecated, and will be removed in 3.0, you should only pass config options. in /var/www/html/project/vendor/php-http/socket-client/src/Client.php on line 59</warning>
> ```
> 
> 1. Beyond the deprecation notice, listing printers also fails due to the following exception:
> 
> ```
> TypeError: Argument 2 passed to Rawilk\Printing\Drivers\Cups\Cups::remoteServer() must be of the type string, null given, called in /var/www/html/project/vendor/rawilk/laravel-printing/src/Factory.php on line 46
> ```
> 
> This error cryptically is trying to say that a username/password are required, but the defaults for the username and password in the published config don't have a default and become null which generates the error.
> 
> Once I set the username/password to be blank strings, I then got the following error:
> 
> ```
> Rawilk\Printing\Exceptions\InvalidDriverConfig with message 'Remote CUPS server requires a username and password.'
> ```
> 
> ~This seems to be due to a arbitrary requirement of needing a username and password for a remote cups server. While not great security wise to have no username/password when I remove the conditional that validates for requiring username/password, the CUPS calls work just fine (albeit with the deprecation notices).~
> 
> ### Steps to reproduce
> Install the package, configure only a valid CUPS_SERVER_IP, run `Printing::printers()`
> Additionally, setting a username/password results in similar errors minus the error on requiring a username/password
> 
> **Context**
> 
> * Laravel Printing version: 1.3
> * Laravel version: 8.35.1

i was getting  `Rawilk\Printing\Exceptions\InvalidDriverConfig with message 'Remote CUPS server requires a username and password.'` too.

The  way I've done is I've initially set `CUPS_SERVER_IP=unset`,  `CUPS_SERVER_USERNAME=unset` & `CUPS_SERVER_PASSWORD=unset`.
This doesn't generate any errors and I don't call any functions if these are unset.

When they are properly configured, i.e: the value is not *unset* then I call the relevant functions.

Hope this helps you :slightly_smiling_face:

**Comment by rawilk** [comment]
I'm not sure if this would be an issue on v2 but feel free to submit a PR that fixes it on v1.x or v2 if still an issue.
